File: fintechplatform/src/main/java/com/fintechplatform/android/account/balance/db/DBBalance.java
package com.fintechplatform.android.account.balance.db;

import com.raizlabs.android.dbflow.sql.language.SQLite;

/**
 * Created by ingrid on 13/02/18.
 */

public class DBBalance {
    public void deleteBalance() {
        SQLite.delete().from(Balance.class).execute();
    }

    public void saveBalance(Balance balance) {
        balance.save();
    }

    public Balance findBalance(String accountId) {
        // TODO search accountId
//        return SQLite.select().from(Balance::class.java).querySingle()
        return SQLite.select()
                .from(Balance.class)
                .where(Balance_Table.accountId.eq(accountId))
                .querySingle();
    }
}

File: dwaplatform/src/main/java/com/dwaplatform/android/account/balance/db/DBBalance.kt
package com.dwaplatform.android.account.balance.db

import com.raizlabs.android.dbflow.sql.language.SQLite

/**
 * Created by ingrid on 07/12/17.
 */
class DBBalance {
    fun deleteBalance() {
        SQLite.delete().from(Balance::class.java).execute()
    }

    fun saveBalance(balance: Balance) {
        balance.save()
    }

    fun findBalance(accountId: String) : Balance? {
        // TODO search accountId
        return SQLite.select().from(Balance::class.java).querySingle()
    }
}
